packages/backend-function/src/factory.ts

```diff
@@ -294,30 +285,36 @@ const nodeVersionMap: Record<NodeVersion, Runtime> = {
   20: Runtime.NODEJS_20_X,
 };
 
-const secretPathSuffix = '_PATH';
-const secretPlaceholderText = '<value will be resolved during runtime>';
 const secretPaths: string[] = [];
 
 const translateEnvironmentProp = (
   functionEnvironmentProp: HydratedFunctionProps['environment'],
   backendSecretResolver: BackendSecretResolver
 ): Record<string, string> => {
+  const secretPlaceholderText = '<value will be resolved during runtime>';
+  const amplifySecretPaths = 'AMPLIFY_SECRET_PATHS';
+  const secretPathEnvVars: Record<string, string> = {};
   const result: Record<string, string> = {};
-  const secretPathEnvVars = [];
 
   for (const [key, value] of Object.entries(functionEnvironmentProp)) {
     if (typeof value !== 'string') {
-      const secretPath = backendSecretResolver.resolveToPath(value);
-      result[key + secretPathSuffix] = secretPath;
+      const secretPath = backendSecretResolver.resolvePath(value);
       result[key] = secretPlaceholderText;
-      secretPathEnvVars.push(key + secretPathSuffix);
+      secretPathEnvVars[key] = secretPath;
       secretPaths.push(secretPath);
     } else {
+      if (key === amplifySecretPaths) {
+        throw new Error(
+          `${amplifySecretPaths} is a reserved environment variable name`
+        );
+      }
       result[key] = value;
     }
   }
 
-  process.env.SECRET_PATH_ENV_VARS = secretPathEnvVars.join(',');
+  if (secretPaths.length > 0) {
+    result[amplifySecretPaths] = JSON.stringify(secretPathEnvVars);
```

Comment:
Is there any harm in including this env var unconditionally?

Reply:
No harm now that I think about it since `secretPathEnvVars` would either have secret data or be `{}`.